### PR TITLE
This is the combined PR for PR3 and PR5

### PR DESCRIPTION
**Pull request by luotigerlsx**

### CODE REVIEW

File: sklearn/sklearn-template/template/trainer/task.py [hunk 1]
```diff
@@ -15,46 +15,99 @@
...
+    object_to_dump: Python object to be pickled
+    output_path: (string) output path which can be Google Cloud Storage
+
+  Returns:
+    None
+  """
+  gcs_path = None
+  parse_result = urlparse(output_path)
 
+  if parse_result.scheme == 'gs':
```
Comment (cfezequiel) [inline sklearn/sklearn-template/template/trainer/task.py]: Sorry, I should've mentioned that `gfile` can handle file I/O for both a local and GCS path and should take care of the 'gs' switching. Considering simplifying.

File: sklearn/sklearn-template/template/setup.py [hunk 2]
```diff
@@ -17,17 +17,16 @@
...
 
-
 REQUIRED_PACKAGES = [
     'matplotlib>=2.2.3',
     'seaborn>=0.9.0',
     'scikit-learn>=0.20.2',
     'pandas-gbq>=0.8.0',
     'cloudml-hypertune',
     'scikit-plot',
+    "urllib",
```
Comment (cfezequiel) [inline sklearn/sklearn-template/template/setup.py]: I understand this was resolved in a previous comment, but looks like the older version got copied to this PR.
To reiterate, suggestion was to replace `"` with `'` for consistency.

**Review by cfezequiel — COMMENTED** [review]
Thanks for combining the PRs. Just a few comments.

File: sklearn/sklearn-template/template/trainer/task.py (hunk 1 above)
Comment (luotigerlsx) [reply sklearn/sklearn-template/template/trainer/task.py]: Modified.

File: sklearn/sklearn-template/template/setup.py (hunk 2 above)
Comment (luotigerlsx) [reply sklearn/sklearn-template/template/setup.py]: the urllib is removed.

**Pull request merged**